src/unstract/api_deployments/client.py

```diff
@@ -141,47 +139,71 @@ def structure_file(self, file_paths: list[str]) -> dict:
         )
         self.logger.debug(response.status_code)
         self.logger.debug(response.text)
-        # The returned object is wrapped in a "message" key. Let's simplify the response.
+        # The returned object is wrapped in a "message" key.
+        # Let's simplify the response.
         obj_to_return = {}
 
-        if response.status_code == 401:
+        try:
+            response_data = response.json()
+            response_message = response_data.get("message", {})
+        except JSONDecodeError:
+            self.logger.error("Failed to decode JSON response", exc_info=True)
```

Comment:
@muhammad-ali-e should we include `response` or `response.text` in case of error for debugging?

Reply:
Updated